controllers/cluster_controller.go

```diff
@@ -185,6 +185,20 @@ func (r *ClusterReconciler) ReconcileUpgrade(ctx context.Context, cluster *clust
 	log.Info(fmt.Sprintf("The cluster CR was upgraded from version %v to %v.", currentVersion, targetVersion))
 	SuccessTotal.WithLabelValues(cluster.Name, cluster.Namespace, currentVersion.String(), targetVersion.String()).Inc()
 	UpgradesInfo.WithLabelValues(cluster.Name, cluster.Namespace).Set(0)
+
+	log.Info("Sending cluster upgrade announcement event.")
```

Comment:
We already send K8s events in `cluster-operator` which are called [`ClusterUpdated`](https://github.com/giantswarm/cluster-operator/blob/master/service/controller/resource/statuscondition/create.go#L193-L200) once we successfully upgraded clusters. 

I think we should use those event in [`event-exporter-app`](https://github.com/giantswarm/event-exporter-app) so it can notify customers once we have finished cluster upgrade. This event is only sent in case we have all the labels set on a rolled nodes and all components are ready.

If we would send the notification here, we wouldn't wait for confirmation from `cluster-operator` verifying the workload cluster.

You could add new a matcher in `event-exporter-app` like [here](https://github.com/giantswarm/event-exporter-app/blob/master/helm/event-exporter-app/templates/configmap.yaml#L24-L30) and add a new receiver to write this [message](https://github.com/giantswarm/upgrade-schedule-operator/pull/20/files#diff-1ca08b6f9e9610ef14189184290a644a29adbb47e6f8653a479d75fdce4a8452R191) for customers 

Example for Matcher:
```
        # We want to send Workload Cluster upgrade events to Slack via the webhook
        # receiver. Therefore we match for events attached to the CAPI Cluster
        # CRs.
        - match:
          - kind: "Cluster"
            reason: "ClusterUpdated"
            receiver: "webhook-successful-upgrade" # or something else

Reply:
All great points.

Is there any benefit in also having a notification of upgrade starting? (Don't want it to be too noisy)

Looking at the cluster-operator, the messages look less user-friendly. Are those suitable for Slack messages to the customers or should we look at different wording?

On a related note, what's the difference between `ClusterUpdated` and `ClusterVersionUpdated`? In what scenarios is a cluster updated without changing version?

Reply:
See https://github.com/giantswarm/event-exporter-app/pull/73

Reply:
> On a related note, what's the difference between ClusterUpdated and ClusterVersionUpdated? In what scenarios is a cluster updated without changing version?

`ClusterVersionUpdated` is a check if all nodes in a workload cluster have the current operator version label set. E.g. if you'd upgrade to a certain release, it would expect the operator version to be present from the release. IIRC this can see it if you would create a new cluster and updating a cluster.

`ClusterUpdated` checks the status of the infrastructure CR, this should be the right one you're looking for, because it's only sent if you would update a cluster. Here's an example from a AWSCluster CR:

```
Status:
  Cluster:
    Conditions:
      Condition:             Updated
      Last Transition Time:  2021-09-21T06:18:15Z
      Condition:             Updating
      Last Transition Time:  2021-09-21T05:00:23Z
      Condition:             Updated
      Last Transition Time:  2021-08-17T06:25:37Z
      Condition:             Updating
      Last Transition Time:  2021-08-17T05:03:40Z
      Condition:             Updated
      Last Transition Time:  2021-06-29T06:47:43Z
      Condition:             Updating
      Last Transition Time:  2021-06-29T05:20:14Z
      Condition:             Updated
      Last Transition Time:  2021-04-27T06:19:40Z
      Condition:             Updating
      Last Transition Time:  2021-04-27T05:10:16Z
      Condition:             Updated
      Last Transition Time:  2020-09-24T06:17:36Z
      Condition:             Updating
      Last Transition Time:  2020-06-10T06:56:26Z
      Condition:             Created
      Last Transition Time:  2020-05-15T10:50:11Z
      Condition:             Creating
      Last Transition Time:  2020-05-15T10:36:51Z
```